zkevm-circuits/src/util.rs

```diff
@@ -34,3 +36,62 @@ pub fn power_of_randomness_from_instance<F: Field, const N: usize>(
 
     power_of_randomness.unwrap()
 }
+
+/// All challenges used in `SuperCircuit`.
+#[derive(Clone, Copy, Debug)]
+pub struct Challenges<T = Challenge> {
+    evm_word: T,
+    keccak_input: T,
+}
+
+impl Challenges {
+    /// Construct `Challenges` by allocating challenges in specific phases.
+    pub fn construct<F: Field>(meta: &mut ConstraintSystem<F>) -> Self {
+        Self {
+            evm_word: meta.challenge_usable_after(FirstPhase),
+            keccak_input: meta.challenge_usable_after(FirstPhase),
+        }
+    }
+
+    /// Get `Expression`s from `Layouter` and returns expressions of challenges.
```

Comment:
We have no layouter here right?
```suggestion
    /// Get `Expression`s from `ConstraintSystem` and returns expressions of challenges.
```

Reply:
Fixed in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/770/commits/96fa4e253365a556fec718ed3550c568f5b4dbd5